src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -28,11 +28,11 @@ QueryToOperatorTransformer::QueryToOperatorTransformer(std::unique_ptr<catalog::
   output_expr_ = nullptr;
 }
 
-OperatorExpression *QueryToOperatorTransformer::ConvertToOpExpression(common::ManagedPointer<parser::SQLStatement> op,
-                                                                      parser::ParseResult *parse_result) {
+std::unique_ptr<OperatorExpression> QueryToOperatorTransformer::ConvertToOpExpression(
+    common::ManagedPointer<parser::SQLStatement> op, parser::ParseResult *parse_result) {
   output_expr_ = nullptr;
   op->Accept(this, parse_result);
-  return output_expr_;
+  return std::move(output_expr_);
```

Comment:
Maybe I'm still recovering from hitting my head, is this std::move(nullptr)??

Reply:
I don't think so. output_expr_ can change after calling Accept.

Reply:
I'm confused where this changes, doesn't come up in a [search](https://github.com/cmu-db/terrier/search?q=output_expr_&unscoped_q=output_expr_). Unless maybe in the optimizer branch?

Reply:
I don't really want to be responsible for logic changes in here. I'm just cleaning up pointers. If logic needs to change I'd open an issue.